tests/ieee802154_submac/main.c

```diff
@@ -195,6 +195,9 @@ static ieee802154_dev_t *_reg_callback(ieee802154_dev_type_t type, void *opaque)
     case IEEE802154_DEV_TYPE_MRF24J40:
         printf("mrf24j40");
         break;
+    case IEEE802154_DEV_TYPE_AT86RF2XX:
```

Comment:
I tried to run the test with `CONTINUE_ON_EXPECTED_ERRORS=1 USEMODULE+=at86rf233 BOARD=nucleo-f767zi`.

I get linking errors:
```
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/netdev_ieee802154_submac/netdev_ieee802154_submac.o: in function `ieee802154_submac_bh_request':
/home/fabian/forks/RIOT/drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c:133: multiple definition of `ieee802154_submac_bh_request'; /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/application_tests_ieee802154_submac/main.o:/home/fabian/forks/RIOT/tests/ieee802154_submac/main.c:174: first defined here
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/netdev_ieee802154_submac/netdev_ieee802154_submac.o: in function `ieee802154_submac_ack_timer_set':
/home/fabian/forks/RIOT/drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c:144: multiple definition of `ieee802154_submac_ack_timer_set'; /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/application_tests_ieee802154_submac/main.o:/home/fabian/forks/RIOT/tests/ieee802154_submac/main.c:134: first defined here
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/netdev_ieee802154_submac/netdev_ieee802154_submac.o: in function `ieee802154_submac_ack_timer_cancel':
/home/fabian/forks/RIOT/drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c:153: multiple definition of `ieee802154_submac_ack_timer_cancel'; /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/application_tests_ieee802154_submac/main.o:/home/fabian/forks/RIOT/tests/ieee802154_submac/main.c:138: first defined here
collect2: error: ld returned 1 exit status
make: *** [/home/fabian/forks/RIOT/tests/ieee802154_submac/../../Makefile.include:740: /home/fabian/forks/RIOT/tests/ieee802154_submac/bin/nucleo-f767zi/tests_ieee802154_submac.elf] Fehler 1
```

Reply:
that board pulls an `netdev_eth`, which selects `netdev`. As a result, the build system pulls `netdev_ieee802154_submac`, which shouldn't be there...

The proper fix for this is https://github.com/RIOT-OS/RIOT/pull/19052. I will try to come up with a workaround

Reply:
This is hard to fix as it is now :/. How should we proceed? Should we try to get #19052?

Reply:
any opinions? @benpicco @fabian18

Reply:
yes, just because we cannot provide netdev and the radio HAL at the same time for IEEE 802.15.4 devices

Reply:
How is the border router going to work then?

Reply:
hmmm wait, it's definitely possible to run the nrf52840 with a border router... I think it hast to do with the way this driver is modeled.
It's still not possible to run the HAL and `netdev_ieee802154` variants at the same time, but it shouldn't affect the BR. I will give it a look.

Reply:
ok, the reason is because the submac callbacks are just C functions that should be defined somewhere. This would only happen with this test application (as `netdev_ieee802154_submac` already implements these callbacks and therefore there won't be duplicated symbols).

So yes, this only occurs in `tests/ieee802154_submac`

Reply:
Not sure if it is a real solution but at least it links fine with `DISABLE_MODULE += netdev_ieee802154_submac` in `tests/ieee802154_submac/Makefile`